### PR TITLE
edit 2.0 public api interface except fluid

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

Others
### PR changes

APIs
### Describe

edit 2.0 public api interface except fluid; resume tensor.math.__all__;tensor.logic.__all__; test=develop

### CODE REVIEW

File: python/.gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 *pyc
 build
+sych.sh
```
Comment (willthefrog) [inline python/.gitignore]: remove this

File: python/paddle/amp/grad_scaler.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 from paddle.fluid.dygraph.amp import AmpScaler
 
-__all__ = ['GradScaler']
+__all__ = ['AmpScaler']
```
Comment (willthefrog) [inline python/paddle/amp/grad_scaler.py]: verify

File: python/paddle/fluid/tests/.gitignore [hunk 3]
```diff
@@ -1,4 +1,51 @@
+.gitignore
```
Comment (willthefrog) [inline python/paddle/fluid/tests/.gitignore]: `.gitignore` files should not be  modified

File: python/paddle/__init__.py [hunk 4]
```diff
@@ -11,6 +11,35 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+__all__ = [
```
Comment (willthefrog) [inline python/paddle/__init__.py]: use one item per line format, add annotations to silent style linter errors if needed,
same for other files

File: python/paddle/distributed/__init__.py [hunk 5]
```diff
@@ -24,10 +24,11 @@
...
 from . import collective
 from .collective import *
+from ..fluid.dygraph.parallel import prepare_context  #DEFINE_ALIAS
 
 from .entry_attr import ProbabilityEntry
 from .entry_attr import CountFilterEntry
-
+'''
 # start multiprocess apis
 __all__ = ["spawn"]
```
Comment (willthefrog) [inline python/paddle/distributed/__init__.py]: verify, this one is shadow by `__all__` bellow?

File: python/paddle/distributed/__init__.py [hunk 6]
```diff
@@ -55,3 +56,10 @@
 
 # collective apis
 __all__ += collective.__all__
+'''
+__all__ = [
```
Comment (willthefrog) [inline python/paddle/distributed/__init__.py]: this one shadows all previous `__all__`s

File: python/paddle/distributed/__init__.py [hunk 7]
```diff
@@ -24,10 +24,11 @@
 
 from . import collective
 from .collective import *
+from ..fluid.dygraph.parallel import prepare_context  #DEFINE_ALIAS
```
Comment (willthefrog) [inline python/paddle/distributed/__init__.py]: verify needed

File: python/paddle/distributed/__init__.py [hunk 8]
```diff
@@ -55,3 +56,10 @@
 
 # collective apis
 __all__ += collective.__all__
+'''
+__all__ = [
+    'spawn', 'init_parallel_env', 'get_rank', 'get_world_size',
+    'InMemoryDataset', 'QueueDataset', 'broadcast', 'all_reduce', 'reduce',
+    'all_gather', 'scatter', 'barrier', 'ReduceOp', 'FileInstantDataset',
```
Comment (willthefrog) [inline python/paddle/distributed/__init__.py]: some of these, e.g.,  `all_reduce` should be include by  `__all__ += collective.__all__`

File: python/paddle/distributed/fleet/base/distributed_strategy.py [hunk 9]
```diff
@@ -19,7 +19,7 @@
 import google.protobuf.text_format
 import google.protobuf
 
-__all__ = ["DistributedStrategy"]
```
Comment (willthefrog) [inline python/paddle/distributed/fleet/base/distributed_strategy.py]: verify

**Review by willthefrog — CHANGES_REQUESTED** [review]
please fix the problem raised first for **all** the files (I only checked the first few, but I suppose other files have similar problems too).

File: python/paddle/amp/grad_scaler.py (hunk 2 above)
Comment (zhiboniu) [reply python/paddle/amp/grad_scaler.py]: GradScaler should access by paddle.amp.GradScaler in 2.0API

File: python/paddle/distributed/__init__.py (hunk 7 above)
Comment (zhiboniu) [reply python/paddle/distributed/__init__.py]: prepare_context should access by paddle.distributed.prepare_context in 2.0API

File: python/.gitignore (hunk 1 above)
Comment (zhiboniu) [reply python/.gitignore]: done

File: python/paddle/fluid/tests/.gitignore (hunk 3 above)
Comment (zhiboniu) [reply python/paddle/fluid/tests/.gitignore]: done

File: python/paddle/nn/__init__.py [hunk 10]
```diff
@@ -22,14 +22,41 @@
 from .utils import weight_norm_hook
 
 from . import initializer
-
+#from .layer.extension import RowConv
+from ..fluid.layers.nn import clip_by_norm  #DEFINEALIAS
```
Comment (willthefrog) [inline python/paddle/nn/__init__.py]: spelling error

File: python/paddle/nn/functional/extension.py [hunk 11]
```diff
@@ -13,16 +13,13 @@
 # limitations under the License.
 
 # TODO: define the extention functions
-
-__all__ = ['diag_embed']
+__all__ = []
```
Comment (willthefrog) [inline python/paddle/nn/functional/extension.py]: remove dead code

File: python/paddle/amp/grad_scaler.py (hunk 2 above)
Comment (willthefrog) [reply python/paddle/amp/grad_scaler.py]: revert this change

**Pull request closed without merging**